src/main.rs

```diff
@@ -175,24 +212,101 @@ fn generate_solid_background(color: Color, background_surface: &mut WlSurface,
 
 /// Given the data from an image, writes it to a special Wayland surface
 /// which is then rendered as a background for Way Cooler.
-fn generate_image_background(path: &str, background_surface: &mut WlSurface,
+fn generate_image_background(path: &str, mode: BackgroundMode, background_surface: &mut WlSurface,
                              env: &WaylandEnv) -> BufferResult {
     // TODO support more formats, split into separate function
-    let image = open(path)
+    let mut image = open(path)
         .unwrap_or_else(|_| {
             println!("Could not open image file \"{:?}\"", path);
             ::std::process::exit(1);
         });
     /*let image = load_from_memory(CURSOR)
         .expect("Could not read cursor data, report to maintainer!");*/
-    let mut image = image.to_rgba();
-    let width = image.width();
-    let height = image.height();
-    let stride = width * 4;
-    let size = stride * height;
+    let mut img_width = image.width();
+    let mut img_height = image.height();
+
+    let dbus_con = Connection::get_private(BusType::Session).unwrap();
+    let resolution = get_screen_resolution(dbus_con);
+    let (scr_width, scr_height) = (resolution.0 as u32, resolution.1 as u32);
+
+    // Mode image processing
+    // The output must be scr_width x scr_height resolution
+    image = match mode {
```

Comment:
Most of these reassignments could be replaced by rebinding (using `let image =` a second time). This is something that Rust lets you do.